bundles/org.openhab.binding.siemensrds/src/main/java/org/openhab/binding/siemensrds/internal/RdsHandler.java

```diff
@@ -312,16 +313,21 @@ private synchronized void doHandleCommand(String channelId, Command command) {
                 if (channelId.equals(channel.id)) {
                     switch (channel.id) {
                         case CHA_TARGET_TEMP: {
-                            Command doCommand = command;
+                            double targetTemperature = Double.NaN;
                             if (command instanceof QuantityType<?> quantityCommand) {
                                 Unit<?> unit = points.getPointByClass(channel.clazz).getUnit();
                                 QuantityType<?> temp = quantityCommand.toUnit(unit);
                                 if (temp != null) {
-                                    doCommand = temp;
+                                    targetTemperature = temp.doubleValue();
                                 }
+                            } else if (command instanceof DecimalType decimal) {
+                                targetTemperature = decimal.doubleValue();
```

Comment:
Very minor, consider naming to be consistent with line 317:
```suggestion
                            } else if (command instanceof DecimalType decimalCommand) {
                                targetTemperature = decimalCommand.doubleValue();
```